src/api/club/APIDetail.ts

```diff
@@ -1,5 +1,35 @@
 import { APIRequest, HTTP_METHOD } from 'interfaces/APIRequest';
-import type { HotClubsResponse } from './entity';
+import type {
+  ClubCategoriesResponse,
+  ClubListResponse,
+  HotClubsResponse,
+} from './entity';
+
+export class ClubCategories<R extends ClubCategoriesResponse> implements APIRequest<R> {
+  method = HTTP_METHOD.GET;
+
+  path = '/clubs/categories';
+
+  response!: R;
+
+  auth = false;
+}
+
+export class ClubList<R extends ClubListResponse> implements APIRequest<R> {
+  method = HTTP_METHOD.GET;
+
+  path = '/clubs';
+
+  response!: R;
+
+  auth = false;
+
+  constructor(public categoryId?: number, public hitSort?: boolean) {
+    this.path = '/clubs'
+    + `${categoryId !== undefined ? `?categoryId=${categoryId}` : ''}`
+    + `${hitSort !== undefined ? `${categoryId !== undefined ? '&' : '?'}hitSort=${hitSort}` : ''}`;
```

Comment:
1. 특정 params가 있는지 확인할때 undefined인지 확인하기보다 그 자체의 값을 확인하는게 좋아보여요.
2. 삼항연산자를 중첩해서 사용했는데 `URLSearchParams`를 사용하거나 이항연산자의 조합으로 변경하면 좀 더 가독성이 좋을 것 같아요!

아래 같은 느낌이 되겠네요!
```suggestion
    + `${categoryId && `?categoryId=${categoryId}` || ''}`
    + `${hitSort && `${categoryId ? '&' : '?'}hitSort=${hitSort}` || ''}`;
```

Reply:
이 부분에 대해서 코드가 더러워보여(?) 고민이 좀 있었는데 좋은 의견 감사합니다~
수정했습니다!